src/main/java/com/radixdlt/engine/RadixEngine.java

```diff
@@ -209,105 +84,92 @@ public void delete(Atom atom) {
 		return constraintMachine.validate(cmAtom.getCMInstruction());
 	}
 
-	// TODO use reactive interface
-	public void store(Atom atom, AtomEventListener atomEventListener) {
+	public void store(Atom atom) throws RadixEngineException {
 		Objects.requireNonNull(atom);
-		Objects.requireNonNull(atomEventListener);
 
 		RadixEngineAtom cmAtom;
 		try {
 			cmAtom = RadixEngineUtils.toCMAtom(atom);
 		} catch (RadixEngineUtils.CMAtomConversionException e) {
 			log.error("Atom creation failed", e);
 			CMError cmError = new CMError(e.getDataPointer(), CMErrorCode.INVALID_PARTICLE, null);
-			atomEventListener.onCMError(atom, cmError);
 			this.atomEventListeners.forEach(acceptor -> acceptor.onCMError(atom, cmError));
-			return;
+			throw new RadixEngineException(RadixEngineErrorCode.CONVERSION_ERROR, e.getDataPointer());
```

Comment:
Would this be better as subclasses of RadixEngineException, rather than instances of RadixEngineException with an embedded RadixEngineErrorCode?